pycqed/measurement/detector_functions.py

```diff
@@ -590,6 +590,13 @@ def poll_data(self):
             t_callback = time.time()
             n_acq_last = [0] * len(self.UHFs)
             n_acq_add = [0] * len(self.UHFs)
+            progress_scaling = 1
+            if hasattr(self.detectors[0], 'nr_shots'):
```

Comment:
we can use self.detectors because it is an attribute of the UHF_Base and is properly defined even for a single-detector-function acquisition. There is now a comment explaining this.